src/bundle/SystemInfo/Collector/ServicesSystemInfoCollector.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace Ibexa\Bundle\SystemInfo\SystemInfo\Collector;
+
+use Ibexa\SystemInfo\Service\ServiceProviderInterface;
+use Ibexa\Bundle\SystemInfo\SystemInfo\Value\ServicesSystemInfo;
+
+/**
+ * Collects information about the services used within the project.
+ */
+final class ServicesSystemInfoCollector implements SystemInfoCollector
+{
+    /** @var \Ibexa\SystemInfo\Service\ServiceProviderInterface */
+    private $serviceProvider;
+
+    public function __construct(ServiceProviderInterface $serviceProvider)
+    {
+        $this->serviceProvider = $serviceProvider;
+    }
+
+    public function collect(): ServicesSystemInfo
+    {
+        return new ServicesSystemInfo([
+            'searchEngine' => $this->serviceProvider->provide('searchEngine')->getValue(),
+            'httpCacheProxy' => $this->serviceProvider->provide('httpCacheProxy')->getValue(),
+            'persistenceCacheAdapter' => $this->serviceProvider->provide('persistenceCacheAdapter')->getValue(),
+        ]);
```

Comment:
Personally I'd prefer a typed constructor instead of generic one provided by `ValueObject`.

This has the added benefit of making sure that all properties of `ServicesSystemInfo` are actually always initialized.

Reply:
I'm ok with injecting an array-map of `string` -> `string`, or even `string` -> `object` if the information provided gets expanded. However, if so, then `ServicesSystemInfo` should not declare a property per single information, but instead keep it under a single, array-based property.

Otherwise there is no benefit.

From my point of view it's a single, basic collector without any thought given towards potential extensibility, and should be treated as such. If we're going with extensibility in mind then the design would change a lot more - it would include more generic templates, maybe more complex potential data structure (as config value is rather crude on it's own - I was already wondering if we could get service class name or service name itself).

I would keep it as-is, only adjusting the constructor, and treat it as internal until we make up our minds about the design. In other words keep it simple until we know what is actually needed.

Reply:
If we aim to introduce better extensibility we would need to take a general approach. Now, all the details presented within System Information are organized in a form of chunks of data (per tab/collector) that are not extensible on their own anyway. I have nothing against doing so, but that would require a follow-up most likely.

Reply:
> From my point of view it's a single, basic collector without any thought given towards potential extensibility, and should be treated as such. If we're going with extensibility in mind then the design would change a lot more - it would include more generic templates, maybe more complex potential data structure (as config value is rather crude on it's own - I was already wondering if we could get service class name or service name itself).
> 
> I would keep it as-is, only adjusting the constructor, and treat it as internal until we make up our minds about the design. In other words keep it simple until we know what is actually needed.

Respecting open-closed principle, while useful for extensibility too, it's rather a general practice, so our internal further scalability of that code is better. But I see that at this point might be an overkill, so I'll drop it for now :)